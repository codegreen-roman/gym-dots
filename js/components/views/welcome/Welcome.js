import React, { Component } from 'react'
import { func, object } from 'prop-types'
import { equals } from 'ramda'
import {
    welcomeSection,
    welcomeHeader,
    welcomeBody,
    buttonsWrapper,
    buttonAndIconWrap,
    buttonText,
    headerDate,
    headerText
} from './Welcome.glamor.js'
import { Icon } from '@components/common/icon/Icon'
import { variables } from '@utils/variables'

export class Welcome extends Component {

    componentWillReceiveProps({status, history}) {
        const isLoggedIn = equals(status, 'loggedIn')
        if (isLoggedIn) {
            history.push('/activity/pre')
        }
    }

    render() {
        const { status, loginGuest, loginWith, dateStr} = this.props
        const isLoggedIn = equals(status, 'loggedIn')
        if (isLoggedIn) {
            return null
        }

        return (
            <section {...welcomeSection}>
                <div {...welcomeHeader}>
                    <div {...headerDate} data-test='date'>{dateStr}</div>
                    <div {...headerText}>Gym dots</div>
                </div>
                <div {...welcomeBody}>
                    <div>Welcome!</div>
                    <div>Login into gym dots systems</div>
                    <div {...buttonsWrapper}>
                        <div {...buttonAndIconWrap}>
                            <Icon icon='facebook' size={30} color={variables.$color2} />
                            <button {...buttonText} data-test='facebook' onClick={() => loginWith('facebook')}>login with facebook</button>
                        </div>
                        <div {...buttonAndIconWrap}>
                            <Icon icon='twitter' size={30} color={variables.$color2} />
                            <button {...buttonText} data-test='twitter' onClick={() => loginWith('twitter')}>login with twitter</button>
                        </div>
                        <div {...buttonAndIconWrap}>
                            <Icon icon='guest' size={30} color={variables.$color2} />
                            <button {...buttonText} data-test='guest' onClick={loginGuest}>login as guest</button>
                        </div>
                    </div>
                </div>
            </section>
        )
    }
}

Welcome.propTypes = {
    loginWith: func.isRequired,
    loginGuest: func.isRequired,
    history: object
}
