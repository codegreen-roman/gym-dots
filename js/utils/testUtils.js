export const findByDataAttr = (wrapper, value) => {
    return wrapper.find(`[data-test='${value}']`)
}
