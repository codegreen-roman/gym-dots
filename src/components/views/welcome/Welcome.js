import React, { PureComponent } from 'react'
import { func, object, string } from 'prop-types'
import { equals } from 'ramda'
import {
    welcomeSection,
    welcomeHeader,
    welcomeBody,
    buttonsWrapper,
    buttonAndIconWrap,
    buttonText,
    headerDate,
    headerText
} from './Welcome.glamor.js'
import { Icon } from '@components/common/icon/Icon'
import { variables } from '@utils/variables'

export class Welcome extends PureComponent {

    static propTypes = {
        loginWith: func.isRequired,
        status: string.isRequired,
        dateStr: string.isRequired,
        loginGuest: func.isRequired,
        history: object
    }

    componentWillReceiveProps({status, history}) {
        const isLoggedIn = equals(status, 'loggedIn')
        if (isLoggedIn) {
            history.push('/in/activity/pre')
        }
    }

    render() {
        const { status, loginGuest, loginWith, dateStr } = this.props
        const isLoggedIn = equals(status, 'loggedIn')
        if (isLoggedIn) {
            return null
        }

        return (
            <section {...welcomeSection}>
                <div {...welcomeHeader}>
                    <div {...headerDate} data-test='date'>{dateStr}</div>
                    <div {...headerText}>Gym dots</div>
                </div>
                <div {...welcomeBody}>
                    <div>Welcome!</div>
                    <div>Login into gym dots systems</div>
                    <div {...buttonsWrapper}>
                        <div {...buttonAndIconWrap}>
                            <Icon
                                iconName='facebook'
                                color={variables.$color2}
                                width={26}
                                height={26}
                            />
                            <button {...buttonText} data-test='facebook' onClick={() => loginWith('facebook')}>login with facebook</button>
                        </div>
                        <div {...buttonAndIconWrap}>
                            <Icon
                                iconName='twitter'
                                color={variables.$color2}
                                viewBox='0 0 20 17'
                                width={23}
                                height={23}
                            />
                            <button {...buttonText} data-test='twitter' onClick={() => loginWith('twitter')}>login with twitter</button>
                        </div>
                        <div {...buttonAndIconWrap}>
                            <Icon
                                iconName='guest'
                                color={variables.$color2}
                                width={26}
                                height={26}
                            />
                            <button {...buttonText} data-test='guest' onClick={loginGuest}>login as guest</button>
                        </div>
                    </div>
                </div>
            </section>
        )
    }
}
