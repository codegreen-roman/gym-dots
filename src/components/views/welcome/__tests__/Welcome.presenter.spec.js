import React from 'react'
import { WelcomePresenter } from '../Welcome.presenter'
import { mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { findByDataAttr } from '@utils/testUtils'

const setup = () => {
    const props = {
        dateStr: 'Saturday, Jul 15th',
        exerciseName: '',
        status: '',
        authWith: jest.fn(),
        authAnonymously: jest.fn()
    }

    const wrapper = mount(<WelcomePresenter {...props} />)

    return {
        wrapper,
        props,
        facebookBtn: wrapper.find('Button[data-test="facebook"]'),
        twitterBtn: wrapper.find('Button[data-test="twitter"]'),
        guestBtn: wrapper.find('Button[data-test="guest"]'),
        dateElement: findByDataAttr(wrapper, 'date')
    }
}

describe('WelcomePresenter component', () => {
    describe('Not logged in', () => {
        it('Contains WelcomeHeader + WelcomeBody', () => {
            const { wrapper } = setup()
            expect(wrapper.find('WelcomeBody').length).toBe(1)
            expect(wrapper.find('WelcomeHeader').length).toBe(1)
        })

        describe('Should contain', () => {
            test('dateElement with a date value Saturday, Jul 15th', () => {
                const { props, dateElement } = setup()
                expect(dateElement.text()).toBe(props.dateStr)
            })

            test('Login btns text should match', () => {
                const { facebookBtn, twitterBtn, guestBtn } = setup()
                expect(facebookBtn.text()).toBe('login with facebook')
                expect(twitterBtn.text()).toBe('login with twitter')
                expect(guestBtn.text()).toBe('login as guest')
            })
        })

        describe('When performing login', () => {
            test('should call the loginWith with twitter provider', () => {
                const { twitterBtn, props } = setup()
                twitterBtn.simulate('click')
                expect(props.authWith).toBeCalledWith('twitter')
            })

            test('should call the loginWith with facebook provider', () => {
                const { facebookBtn, props } = setup()
                facebookBtn.simulate('click')
                expect(props.authWith).toBeCalledWith('facebook')
            })

            test('should call the loginGuest', () => {
                const { guestBtn, props } = setup()
                guestBtn.simulate('click')
                expect(props.authAnonymously).toHaveBeenCalled()
            })
        })

        describe('Shallow snapshot', () => {
            test('matches the previous', () => {
                const component = setup()
                const tree = shallowToJson(component)
                expect(tree).toMatchSnapshot()
            })
        })
    })
})
