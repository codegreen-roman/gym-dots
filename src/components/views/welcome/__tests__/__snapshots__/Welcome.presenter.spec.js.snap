// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomePresenter component Not logged in Shallow snapshot matches the previous 1`] = `null`;
